nav2_behavior_tree/test/plugins/decorator/test_distance_controller.cpp

```diff
@@ -57,34 +56,31 @@ class DistanceControllerTestFixture : public ::testing::Test
   static void TearDownTestCase()
   {
     transform_handler_->deactivate();
-    delete transform_handler_;
-    delete config_;
-    delete dummy_node_;
-    transform_handler_ = nullptr;
-    config_ = nullptr;
-    dummy_node_ = nullptr;
   }
 
   void SetUp()
   {
     node_ = new nav2_behavior_tree::DistanceController("distance_controller", *config_);
```

Comment:
Was this deleted?

Reply:
There was a suggestion earlier about using smart pointers in these tests wherever possible. I added that to the speed controller test so I ended up changing the distance controller test as well.

Reply:
This isn't a smart pointer though, this is a raw pointer if you use `new`. If you use `new` you must delete it or you're leaking memory. Making these all smart pointers I think is the best call, but honestly I don't really care because its a test, it just needs to be properly deleted so we don't leak at bare minimum.

Reply:
Ah yes I misunderstood what you meant by deleted. Should be okay now.